File: 0001-two_sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:    
        hashmap = {}
        for i in range(len(nums)):
          diff = target - nums[i]

          if nums[i] in hashmap:
            return [hashmap[nums[i]], i]

          hashmap[diff] = i

        return []

File: 0001-two_sum.go
func twoSum(nums []int, target int) []int {
	seen := make(map[int]int)

	for i, num := range nums {
		diff := target - num
		if j, ok := seen[diff]; ok {
			return []int{j, i}
		}

		seen[num] = i
	}

	return []int{}
}